packages/cpp/src/browser/cpp-language-client-contribution.ts

```diff
@@ -52,21 +59,22 @@ export class CppLanguageClientContribution extends BaseLanguageClientContributio
     protected onReady(languageClient: ILanguageClient): void {
         super.onReady(languageClient);
 
-        // When the language server is ready, send the active build
-        // configuration so it knows where to find the build commands
-        // (e.g. compile_commands.json).
-        this.onActiveBuildConfigChanged(this.cppBuildConfigurations.getActiveConfig());
         this.cppBuildConfigurations.onActiveConfigChange(config => this.onActiveBuildConfigChanged(config));
 
         // Display the C/C++ build configurations status bar element to select active build config
         this.cppBuildConfigurationsStatusBarElement.show();
     }
 
+    private createClangdConfigurationParams(config: CppBuildConfiguration | undefined): ClangdConfigurationParamsChange {
+        const clangdParams: ClangdConfigurationParamsChange = {
+            compilationDatabasePath: config ? config.directory : undefined
```

Comment:
See PR #2437, I think we'll want to use `""` instead of `undefined`.

Reply:
We'll have to make sure using "" doesn't disable the fallback "check in the parent" algorithm.